core/src/main/java/org/openconnectors/connect/SourceContext.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.openconnectors.connect;
+
+import org.openconnectors.config.Config;
+
+import java.io.Closeable;
+import java.io.Serializable;
+import java.util.Collection;
+
+public interface SourceContext<T> extends Closeable, Serializable {
+
+    void setup(Config config) throws Exception;
```

Comment:
Could you describe what setup/collect are?

Reply:
refer to docs or a sample implementation
https://github.com/openconnectors/connectors-legacy-prototype/blob/master/java/core/src/main/java/org/streamlio/localfs/LocalFSSource.java

Reply:
Would it be better if SourceContext will implement `ConnectorContext`? The Sink does it as i see it below.

Reply:
SourceContext serves a different purpose then SourceConnectorContext. It probably needs a better name